segmentstore/server/host/src/main/java/io/pravega/segmentstore/server/host/handler/PravegaRequestProcessor.java

```diff
@@ -1048,14 +1047,7 @@ private void truncateIndexSegment(String segment, long offset) {
             if (indexSegmentOffset == 0) {
                 return;
             }
-            segmentStore.truncateStreamSegment(indexSegment, indexSegmentOffset, TIMEOUT)
-                    .whenComplete((v, exp) -> {
-                        if (exp != null) {
-                            log.warn("Failed to truncate index segment {} at offset {} due to", indexSegment, indexSegmentOffset, exp);
-                        } else {
-                            log.info("Successfully truncated index segment {} at offset {}", indexSegment, indexSegmentOffset);
-                        }
-                    });
+            segmentStore.truncateStreamSegment(indexSegment, indexSegmentOffset, TIMEOUT);
         } catch (Exception e) {
             log.warn("Unable to locate offset for index segment {}  for offset {} due to ", indexSegment, offset, e);
```

Comment:
What will be the result if this exception occurs ?

Reply:
If the exception occurs then I have to throw the respective exception to its caller, so that it will retry.

Reply:
I don't think this exception should be thrown to caller